services/slam/server_test.go

```diff
@@ -133,6 +147,19 @@ func TestServer(t *testing.T) {
 		test.That(t, resp, test.ShouldBeNil)
 	})
 
+	t.Run("failing get internal state function", func(t *testing.T) {
+		injectSvc.GetInternalStateFunc = func(ctx context.Context, name string) ([]byte, error) {
+			return nil, errors.New("failure to get internal state")
+		}
+
+		req := &pb.GetInternalStateRequest{
+			Name: testSlamServiceName,
+		}
+		resp, err := slamServer.GetInternalState(context.Background(), req)
+		test.That(t, err, test.ShouldNotBeNil)
```

Comment:
Should this error be the same error from here? `errors.New("failure to get internal state")` or otherwise able to be verified with equality, rather than just not nil?